packages/desktop-client/src/components/sidebar/Account.tsx

```diff
@@ -87,12 +102,52 @@ export function Account<FieldName extends SheetFields<'account'>>({
     onDrop,
   });
 
+  const dispatch = useDispatch();
+
+  const editingRef = useRef(null);
+  const [isEditing, setIsEditing] = useState(false);
+  useEffect(() => {
+    if (!editingRef.current) return;
+    if (isEditing) {
+      editingRef.current.focus();
+      window.getSelection().selectAllChildren(editingRef.current);
+    } else {
+      editingRef.current.textContent = name;
+    }
+  }, [name, isEditing]);
+
+  const updateName = () => {
+    if (account && isEditing) {
+      setIsEditing(false);
+      const newName = editingRef.current.textContent;
+      if (newName !== account.name && newName.trim()) {
+        dispatch(
+          updateAccount({
+            ...account,
+            name: newName,
+          }),
+        );
+      }
+    }
+  };
+
   const accountNote = useNotes(`account-${account?.id}`);
   const needsTooltip = !!account?.id;
 
   const accountRow = (
-    <View innerRef={dropRef} style={{ flexShrink: 0, ...outerStyle }}>
-      <View>
+    <View
+      innerRef={dropRef}
+      style={{ flexShrink: 0, ...outerStyle }}
+      onContextMenu={e => {
+        if (!needsTooltip || !contextMenusEnabled) return;
+        e.preventDefault();
+        const rect = e.currentTarget.getBoundingClientRect();
+        setCrossOffset(e.clientX - rect.left);
```

Comment:
This logic seems the same across a lot of the context menus? Is it possible to put the logic in a hook and reuse it instead of copying and pasting it for each menu?

Reply:
I have never created hooks with react so no clue, I will try.

Reply:
@matt-fidd thanks! I added this to https://github.com/actualbudget/actual/pull/3775 as it seemed as the more approriate PR.

[`864251d` (#3775)](https://github.com/actualbudget/actual/pull/3775/commits/864251db927a45fe7cdf31a7e54afb888a11dbfd)

Reply:
Can you rebase this on top of the other PR so it uses the hook?